caffe2/core/dispatch/DeviceId.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <functional>
+#include <iostream>
+
+namespace c10 {
+
+enum class DeviceTypeId : uint8_t {
```

Comment:
This needs comments saying under what situations its extensible, and how to extend it. Additionally, there should be some guidance about whether or not you're allowed to switch() on it.